### PR TITLE
Remove outdated CoreFx.Private.TestUtilities.Unicode references

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
There remained some stray references after a recent refactoring.

I also updated the solution file to make the newly added _System.Globalization.Nls.Tests_ project depend on the src project, just like the old _System.Globalization.Tests_ project does.

### CODE REVIEW

File: src/libraries/System.Globalization/System.Globalization.sln [hunk 1]
```diff
@@ -26,6 +26,9 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "ref", "ref", "{2E666815-2EDB-464B-9DF6-380BF4789AD4}"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "System.Globalization.Nls.Tests", "tests\NlsTests\System.Globalization.Nls.Tests.csproj", "{6B71E284-DA57-4657-9E08-A02BAFB877CC}"
+	ProjectSection(ProjectDependencies) = postProject
```
Comment (safern) [inline src/libraries/System.Globalization/System.Globalization.sln:29]: Is this needed?

Comment (GrabYourPitchforks) [reply src/libraries/System.Globalization/System.Globalization.sln:29]: I believe so. All other test projects have this same annotation such that they take a build ordering dependency on the src project.

**Review by safern — COMMENTED** [review]
Thanks for fixing this

**Review by safern — APPROVED** [review]

File: src/libraries/System.Globalization/System.Globalization.sln (hunk 1 above)
Comment (safern) [reply src/libraries/System.Globalization/System.Globalization.sln:29]: I see, yeah makes sense. Do we need to do the same for System.Runtime.Nls.Tests? in System.Runtime.sln?

**Comment by safern** [comment]
FYI: @cshung

File: src/libraries/System.Globalization/System.Globalization.sln (hunk 1 above)
Comment (GrabYourPitchforks) [reply src/libraries/System.Globalization/System.Globalization.sln:29]: Ah, yeah, good catch!

**Comment by GrabYourPitchforks** [comment]
@safern Per your comment at https://github.com/dotnet/runtime/pull/34981#discussion_r408533751 I checked _System.Runtime.sln_ and didn't see a reference at all to the NLS test project. So the most recent commit in this PR adds it.

**Review by safern — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**